src/interface/lazy.jl

```diff
@@ -709,10 +709,33 @@ function compute_parse(ctx, args::Tuple)
     return ress
 end
 
-function Base.argmax(arr::LazyTensor, dims::Real=2)
-    return argmax(arr, dims)
-end
+function Base.argmax(A::AbstractTensor; dims=:)
+    return compute(argmax(lazy(A), dims=dims))
+end
+
+function Base.argmax(A::LazyTensor; dims=:)
+    dims = dims == Colon() ? (1:ndims(A)) : collect(dims)
+
+    if (ndims(A) >= 2)
+        A1 = (map(
+            x -> x[2], 
+            reduce(
+                maxby, 
+                map(
+                    Pair, 
+                    A, 
+                    CartesianIndices(size(A))), 
+                    dims=dims, 
+                    init=-Inf=>CartesianIndex(fill(0, length(size(A)))...)
+            )
+        ))
+        # NOTE: CURRENTLY DOES NOT WORK
+        B = expanddims(A1, dims)
```

Comment:
There's a really odd bug where this does not work in the Lazy function. (It's called and updates the matrix but then it reverts when returned for some reason?) It does work when I pull this into the non-lazy version. Not entirely sure what's up here.